dwave/plugins/sklearn/transformers.py

```diff
@@ -0,0 +1,325 @@
+    ) -> None:
+        """
+        Initialize a hybrid quantum-classical feature selection transformer.
+
+        Args:
+            alpha (float, optional): hyper-parameter which weights correlation to outcome vs covariance. Must be between 0 and 1. Defaults to .5.
+            time_limit (int, optional): a time limit for the Leap Constrained Quadratic Model Solver to solve the feature selection problem.
+            Must be at least the minimum time limit of the solver. Defaults to 10.
+            n_default_features (int, optional)
+            method (string, optional): The method used to calculate the model matrix. One of "correlation" or "mutual information". Defaults to correlation.
+        """
+        super().__init__()
+        self.alpha = alpha
+        self.time_limit = time_limit
+        self.n_default_feature = n_default_feature
+        self.selected_columns = None
+        self.mask = None
+        if (self.alpha > 1) or (self.alpha < 0):
+            raise ValueError(f"alpha {self.alpha} is not between 0 and 1")
+
+        if self.time_limit <= 1:
+            raise ValueError("Time limit must be positive and greater than 1")
+
+        if method not in self.acceptable_methods:
+            raise ValueError(
+                f"method was {method}, must be one of {SelectFromQuadraticModel.acceptable_methods}"
+            )
+
+        self.method = method
+
+    def _get_support_mask(self):
+        """
+        Get the boolean mask indicating which features are selected
+        Returns
+        -------
+        support : boolean array of shape [# input features]
+            An element is True iff its corresponding feature is selected for
+            retention.
+        """
+        if self.mask is None:
+            raise RuntimeError("fit hasn't been run yet")
+
+        return self.mask
+
+    def calculate_correlation_matrix(
+        self, X: np.ndarray, y: Union[np.ndarray, None] = None, chunksize: int = 500
+    ):
+        """_summary_
+
+        Args:
+            X (_type_): _description_
+            y (_type_, optional): _description_. Defaults to None.
+            chunksize (int, optional): _description_. Defaults to 500.
+
+        Returns:
+            _type_: _description_
+        """
+        logging.info(
+            "Starting correlation calculation"
+        )  
+        # heavy logging to diagnose memory performance
+        # generate correlation matrix, use tempfile and chunked process if too big
+        if X.shape[1] < chunksize:
+            logging.info("Using numpy corrcoef")
+            correlation_matrix = np.corrcoef(X, rowvar=False)
+            X_standardized = None
+        else:
+            logging.info("data is too large, will be chunked")
+
+            with tempfile.NamedTemporaryFile() as f:
+                X_base = np.memmap(f, "float64", mode="w+", shape=X.shape)
+
+            if isinstance(X, pd.DataFrame):
+                X_base[:] = X.to_numpy()
```

Comment:
I would drop the pandas dependency altogether. And just rely on the user to do this from the outside. We can still use pandas in the tests.